pkg/reconciler/gc/controller.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2019 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gc
+
+import (
+	"context"
+
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	configurationinformer "knative.dev/serving/pkg/client/injection/informers/serving/v1alpha1/configuration"
+	revisioninformer "knative.dev/serving/pkg/client/injection/informers/serving/v1alpha1/revision"
+	pkgreconciler "knative.dev/serving/pkg/reconciler"
+	configns "knative.dev/serving/pkg/reconciler/configuration/config"
+)
+
+const (
+	controllerAgentName = "gc-controller"
+)
+
+// NewController creates a new Garbage Collection controller
+func NewController(
+	ctx context.Context,
+	cmw configmap.Watcher,
+) *controller.Impl {
+
+	configurationInformer := configurationinformer.Get(ctx)
+	revisionInformer := revisioninformer.Get(ctx)
+
+	c := &reconciler{
+		Base:                pkgreconciler.NewBase(ctx, controllerAgentName, cmw),
+		configurationLister: configurationInformer.Lister(),
+		revisionLister:      revisionInformer.Lister(),
+	}
+	impl := controller.NewImpl(c, c.Logger, "Garbage Collection")
+
```

Comment:
You aren't enqueuing any events, so this controller won't do anything :(

Reply:
Something like: https://github.com/knative/serving/blob/52a6e2c08b979b91693ab4503205721f4b30c9be/pkg/reconciler/configuration/controller.go#L51-L57

Reply:
@mattmoor I see. In this case, I don't think this controller needs to respond to any specific configuration or revision events, because we just want to sync globally every few hours. Does the following do that?
```
	configStore := configns.NewStore(c.Logger.Named("config-store"), controller.GetResyncPeriod(ctx))
	configStore.WatchConfigs(c.ConfigMapWatcher)
```

Reply:
No, that just registers callbacks when the configmap changes to update the store.

You need something like this: https://github.com/knative/serving/blob/74e366a72c7f7186a89091df3ec5858c387b5037/pkg/reconciler/revision/controller.go#L111-L117

It looks like to add global resync will require a change to the store to expose `onAfterStore` like this: https://github.com/knative/serving/blob/74e366a72c7f7186a89091df3ec5858c387b5037/pkg/reconciler/revision/config/store.go#L57